### PR TITLE
Generate types in their own folder

### PR DESCRIPTION
**Pull request by diegograsso**

### CODE REVIEW

File: modules/ipfs-cpinner-client/webpack.config.js [hunk 1]
```diff
@@ -18,6 +18,9 @@ module.exports = {
   output: {
     filename: 'bundle.js',
     path: path.resolve(__dirname, 'dist'),
-    libraryTarget: 'umd'
+    libraryTarget: 'umd',
+    library: "RDataVault",
```
Comment (ilanolkies) [inline modules/ipfs-cpinner-client/webpack.config.js:22]:
```suggestion
    library: "RIFDataVault",
```

File: modules/ipfs-cpinner-service/package.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "description": "RIF Identity - IPFS Centralized Pinner Service",
   "private": true,
   "main": "lib/index.js",
+  "types": "lib/index.d.ts",
```
Comment (ilanolkies) [inline modules/ipfs-cpinner-service/package.json:7]: Unnecessary?

File: modules/ipfs-cpinner-client-types/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { IAuthManager as IAuthManagerType, IAuthManagerNewable as IAuthManagerNewableType } from './auth-manager/types'
```
Comment (ilanolkies) [inline modules/ipfs-cpinner-client-types/src/index.ts:1]: Does `export * from '…'` work?

**Review by ilanolkies — CHANGES_REQUESTED** [review]

File: modules/ipfs-cpinner-client-types/src/index.ts (hunk 3 above)
Comment (ilanolkies) [reply modules/ipfs-cpinner-client-types/src/index.ts:1]: This will avoid using `/lib/zzz`

Comment (diegograsso) [reply modules/ipfs-cpinner-client-types/src/index.ts]: Replaced by 
```import { IAuthManager } from './auth-manager/types'
import { IEncryptionManager } from './encryption-manager/types'
export * from './auth-manager/types'
export * from './encryption-manager/types'```

File: modules/ipfs-cpinner-service/package.json (hunk 2 above)
Comment (diegograsso) [reply modules/ipfs-cpinner-service/package.json]: yes, removed

**Review by ilanolkies — APPROVED** [review]

**Pull request merged**